src/server.ts

```diff
@@ -197,9 +201,46 @@ connection.onDidChangeConfiguration((change) => {
 	}
 });
 
+function setSchemaStoreSettingsIfNotSet(){
+	if(!hasSchemaStoreBeenAdded){
+		getSchemaStoreMatchingSchemas().then(schemaStore => {
+			schemaStoreSettings = schemaStore.schemas;
+			updateConfiguration();
+		});
+		hasSchemaStoreBeenAdded = true;
+	}
+}
+
+function getSchemaStoreMatchingSchemas(){
+	
+	return xhr({ url: "http://schemastore.org/api/json/catalog.json" }).then(response => {
+		
+		let languageSettings= {
+			schemas: []
+		};
+
+		let schemas = JSON.parse(response.responseText);
+		for(let schemaIndex in schemas.schemas){
+			
+			let schema = schemas.schemas[schemaIndex];
+			if(schema && schema.name && schema.name.indexOf('.yml') !== -1 || schema.name.indexOf('.yaml') !== -1){

```

Comment:
The name has no semantic meaning, you should look only look at the `fileMatch` property